src/main/java/org/entando/plugins/pda/engine/EngineFactory.java

```diff
@@ -0,0 +1,34 @@
+package org.entando.plugins.pda.engine;
+
+import org.entando.plugins.pda.core.engine.Engine;
+import org.entando.plugins.pda.core.engine.FakeEngine;
+import org.entando.plugins.pda.core.exception.EngineNotFoundException;
+import org.entando.plugins.pda.pam.engine.KieEngine;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+public class EngineFactory implements ApplicationContextAware  {
+    private Map<String, Engine> availableEngines = new HashMap<>();
+
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
```

Comment:
This should be pluggable and injected rather than manually adding the engine implementations.  One way to do this is to do an 

```
@Autowired 
List<Engine> availableEngines
```

So that you automatically get all Engine implementations that exist on the classpath and you don't have to explicitly code them in. This would also allow us to selectively include implementations and only deploy the engines we want.

Reply:
Yes, makes sense. I thought about this, but wasn't finding an easy solution to inject without direct access.